stellar-multitenant/pkg/tenant/manager.go

```diff
@@ -53,32 +56,67 @@ func (m *Manager) GetDSNForTenant(ctx context.Context, tenantName string) (strin
 	return router.GetDSNForTenant(dataSourceName, tenantName)
 }
 
+func (m *Manager) GetDSNForTenantByID(ctx context.Context, id string) (string, error) {
+	t, err := m.GetTenantByID(ctx, id)
+	if err != nil {
+		return "", err
+	}
+
+	return m.GetDSNForTenant(ctx, t.Name)
+}
+
 var selectQuery string = `
 	SELECT 
 		*
 	FROM
 		tenants t
-	%s
 `
 
 // GetAllTenants returns all tenants in the database.
 func (m *Manager) GetAllTenants(ctx context.Context) ([]Tenant, error) {
-	tnts := []Tenant{}
-
-	query := fmt.Sprintf(selectQuery, "ORDER BY t.name ASC")
+	queryParams := &QueryParams{
+		Filters: map[FilterKey]interface{}{
+			FilterKeyOutStatus: DeactivatedTenantStatus,
+		},
```

Comment:
Can you add an optional `filters` parameter, so callers can request the tenants with custom (or without any) filters?
* If they send nil, you replace by the default one already in place
* If they send anything else, you use whatever they sent, including `map[FilterKey]interface{}{}`